shell/src/main/java/alluxio/cli/UnderFileSystemContractTest.java

```diff
@@ -84,6 +109,63 @@ private void run() throws Exception {
     System.out.printf("Tests completed with %d failed.%n", failedCnt);
   }
 
+  /**
+   * Runs the tests and return a {@link alluxio.cli.ValidateUtils.TaskResult}.
+   *
+   * @return a task result for all UFS tests
+   * */
+  public ValidateUtils.TaskResult runValidationTask() throws IOException {
+    Closer closer = Closer.create();
+    final ByteArrayOutputStream msgBuf = new ByteArrayOutputStream();
+    final ByteArrayOutputStream adviceBuf = new ByteArrayOutputStream();
+    PrintStream msgStream = new PrintStream(msgBuf, true);
+    PrintStream adviceStream = new PrintStream(adviceBuf, true);
+    closer.register(msgStream);
+    closer.register(adviceStream);
+    closer.register(msgBuf);
+    closer.register(adviceBuf);
+    try {
+      UnderFileSystemConfiguration ufsConf = getUfsConf();
+      UnderFileSystemFactory factory = UnderFileSystemFactoryRegistry.find(mUfsPath, ufsConf);
+      // Check if the ufs path is valid
+      if (factory == null || !factory.supportsPath(mUfsPath)) {
+        msgStream.append(String.format("%s is not a valid path%n", mUfsPath));
+        adviceStream.append(String.format("Please validate if %s is a correct path\n", mUfsPath));
+        return new ValidateUtils.TaskResult(ValidateUtils.State.FAILED, TASK_NAME,
+                msgBuf.toString(), adviceBuf.toString());
+      }
+
+      // Set common properties
+      mConf.set(PropertyKey.UNDERFS_LISTING_LENGTH, "50");
```

Comment:
Those properties are inherited from original UfsContrctTests and are added to for testing purposes. Like listing less files , using small blocks and speed up tests. I think it will be better to keep these properties